src/Gui/Tree.cpp

```diff
@@ -5441,6 +5441,11 @@ void DocumentObjectItem::testStatus(bool resetStatus, QIcon& icon1, QIcon& icon2
         icon = object()->mergeColorfulOverlayIcons(icon);
 
         if (isVisibilityIconEnabled()) {

```

Comment:
```suggestion
        if (isVisibilityIconEnabled() && strlen(object()->DisplayMode.getValueAsString())) {
```

I think this line is enough to filter the switchable objects.

Reply:
Adding the filter here messes up the alignment.
Checking display mode might be interesting though :)